### PR TITLE
RUN-4512 Incremental change to organize elipc send responsibilities

### PR DESCRIPTION
**Pull request by StevenEBarbaro**
Split logical responsibilities in elipc send. These changes are an incremental step towards message batching.

https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5b8ecc436107b45e6d1ebc05

### CODE REVIEW

File: src/browser/api_protocol/transport_strategy/elipc_strategy.ts [hunk 1]
```diff
@@ -50,6 +50,32 @@ export class ElipcStrategy extends ApiTransportBase<MessagePackage> {
         });
     }
 
+    private canTrySend(routingInfo: any): boolean {
```
Comment (HarsimranSingh) [inline src/browser/api_protocol/transport_strategy/elipc_strategy.ts]: :neckbeard: should we finally make a type for this?

File: src/browser/api_protocol/transport_strategy/elipc_strategy.ts [hunk 2]
```diff
@@ -50,6 +50,32 @@ export class ElipcStrategy extends ApiTransportBase<MessagePackage> {
...
+        const browserWindowExists = !browserWindow.isDestroyed();
+        const validRoutingId = typeof frameRoutingId === 'number';
+        return browserWindowLocated && browserWindowExists && validRoutingId;
+    }
+
+    // Dispatch a message
+    private innerSend(payload: string,
+                      frameRoutingId: number,
+                      mainFrameRoutingId: number,
+                      browserWindow: any): void {
```
Comment (HarsimranSingh) [inline src/browser/api_protocol/transport_strategy/elipc_strategy.ts]: Can this be typed `BrowserWindow`?

**Comment by rdepena** [comment]
🌳 pushed to .36.

File: src/browser/api_protocol/transport_strategy/elipc_strategy.ts [hunk 3]
```diff
@@ -50,6 +50,32 @@ export class ElipcStrategy extends ApiTransportBase<MessagePackage> {
...
+        if (frameRoutingId === mainFrameRoutingId) {
+            // this is the main window frame
+            if (coreState.argo.framestrategy === 'frames') {
```
[the remark this answers is not in the document]
Comment (datamadic) [reply src/browser/api_protocol/transport_strategy/elipc_strategy.ts]: Can we remove the check for `coreState.argo.framestrategy === 'frames'` it is no longer needed and in this case will cause incorrect behavior. Just removing the check is fine, nothing else has to change

[the remark this answers is not in the document]
Comment (StevenEBarbaro) [reply src/browser/api_protocol/transport_strategy/elipc_strategy.ts]: Develop and 35/staging both contain the _if block_ in question

https://github.com/HadoukenIO/core/blob/develop/src/browser/api_protocol/transport_strategy/elipc_strategy.ts#L65

https://github.com/HadoukenIO/core/blob/9.61.35/staging/src/browser/api_protocol/transport_strategy/elipc_strategy.ts#L65

[the remark this answers is not in the document]
Comment (datamadic) [reply src/browser/api_protocol/transport_strategy/elipc_strategy.ts]: will be fixed in a follow up, thanks!

**Pull request merged**